### ISSUE TITLE
about 0.4

### ISSUE THREAD

**Issue by danywheeler** [meanjs/mean#402]
I've been watch 0.4 version for a while , and it's been few month since the release note . I'm wondering when the documentation would be update , and when the 0.4 would be merge to  the master .  I want to use 0.4 as a start point to study and develop project .  by the way nice work with 0.4

**Comment by sylvainlap** [comment]
:+1:

**Comment by ilanbiala** [comment]
@DanyWheeler @sylvainlap 0.4 can be used, it just hasn't been merged into master. If you have any questions, you can open an issue and we'll try to help you out. As for merging it in, I'd have to defer that to someone more senior. @lirantal @amoshaviv @roieki may be able to provide some info about merging 0.4 in and making it official.

**Comment by lirantal** [comment]
@ilanbiala I'm definitely tracking the project and trying to remain as responsive as possible, though I can't say I've got 0.4 pushed further on my calendar (busy time for me atm). It may only seem so, but it looks like @amoshaviv and @roieki are quite busy with their own stuff as well.

Let's see if we can get here some other active contributors whom are willing to work on 0.4 and it's merge to master, and take it from there.

**Comment by miguelcoba** [comment]
Is there a thread about this release process, with guidelines and objectives, so we can contribute also to the release process?

**Comment by ilanbiala** [comment]
@miguelcoba I haven't heard much from @amoshaviv or @roieki about what needs to be finished, so I'm not too sure. Once we hear from them, we'll definitely create a tracking issue.

**Comment by SOSANA** [comment]
Let's get this going... Been waiting a few months on this issue, we need to get the ball rolling.

**Comment by ilanbiala** [comment]
@SOSANA 0.4 can be used from the Git repo right now, but the branch hasn't officially been merged into master.

**Comment by wesleyfsmith** [comment]
@ilanbiala But haven't there been a lot of new changes on master that need to be merged in with the 4.0 branch? That's what concerns me.

It's very important for me to see that meanjs continues to get supported, let me know what I can do to help.

**Comment by ilanbiala** [comment]
@wesleyfsmith the biggest thing at this time that anyone can do is give feedback on issues and PRs so @meanjs/contributors know what people want to see next past 0.4.

**Comment by lirantal** [comment]
we should indeed get 0.4 branch rolled out into master, but this requires quite some effort, including updating docs and such on the meanjs.org website too.

**Comment by miguelcoba** [comment]
I can help with that, if you want, that's why I asked if there is a
guidelines and objectives doc.
As of now, there is a little of slowdown as we wait for core committers to
chime in.
How can I help?

**Comment by ilanbiala** [comment]
@miguelcoba the big things I want to make sure get merged into 0.4 are #343 and #253. It would be great if you can comment with any feedback there so we can get those merged in and 0.4 can be finalized with regards to everything else.

**Comment by trevis-baker** [comment]
I do agree.  What else needs to be done, so the 0.4 branch can be merged to master?  My fear is the activity in this community slowdown as we delay.  I can help.

**Comment by ilanbiala** [comment]
0.4 now has a milestone with some issues in it. It would be great to get feedback on any of the issues that are marked for discussion, and then we can decide on a good solution.

**Comment by lirantal** [comment]
@ilanbiala @mleanos and anyone else wanting to help merge 0.4 into master, lets take this discussion further.

with regards to 0.4.0 branch, it doesn't seem to make sense to merge 'master' into '0.4.0', which means we have to do the other way around and that is to merge '0.4.0' into 'master', resolve all conflicts and push to the repo.

Thing is, even if we do that today, then we have all those PR requests pending on 0.4.0, what should we do? does it sound reasonable to merge '0.4.0' to 'master' as suggested and then accept the PRs on '0.4.0' branch and then re-merge '0.4.0' to 'master' ?

**Comment by mleanos** [comment]
I'm a bit torn on the best strategy to handle this. It's such a big "merge". I recently migrated my app to the 0.4.0. So I have a little bit of a sense, as to the complexity of merging master with 0.4.0. 

How much of what is in master, is critical to merge with 0.4.0? I'm sure there are some bug fixes, dependency differences, and perhaps other features that have been merged into master lately. 

For me it would make most sense to identify what is necessary to pull from master into 0.4.0, and take that approach; as described in the latter part of your last comment. After all, 0.4.0 has most of the desired updates. The master branch shouldn't contain as many enhancements, that aren't already in 0.4.0. Although, I'm not sure if that's even possible to determine.

I say master --> 0.4.0 --> merge PR's ---> 0.4.0 ---> master, is the most logical flow.

**Comment by ilanbiala** [comment]
@lirantal didn't we set up PRs and other commits directly to 0.4.0 so we could just choose that as the copy to keep when merging?

**Comment by lirantal** [comment]
@mleanos @ilanbiala glad to have you guys on-board here :)
@roieki @SOSANA @miguelcoba @wesleyfsmith we'll appreciate if you guys can please comment and refer here to the status of 0.4.0 in my following summary:

I jumped the gun and this is what I did:
1. created master_to_0.4.0 branch out of master (the name may sound confusing)
2. merged 0.4.0 into master
3. lots of conflicts that needs resolving

I'm done resolving conflicts.

It was a long night but I'm mostly after it and I can't say that everything works yet as I didn't do much testing, or all the features in master were copy&paste because I mostly gave reference to 0.4.0 files but I think we have a repository which has 0.4.0 merged into master branch now.

Next steps IMO are:
1. I will run some quick sanity/tests to make sure everything is ok
2. if (1) works without much problem I can push the repository so you guys (and others) can further take a look and if you wish fix/backport any missing items there.
3. I'll then merge master_to_0.4.0 branch into 0.4.0
4. Then we can try to merge all out-standing PR on 0.4.0 into 0.4.0
5. Now merging 0.4.0 into master and tagging master as the 0.4.0 release shouldn't be a problem and no conflicts should arise.

What do you think?

**Comment by mleanos** [comment]
This sounds like a good plan to me. Looking forward to helping out here.

@lirantal Thanks for getting this going. Still a long road ahead of us, but it will be worth the long nights.

**Comment by lirantal** [comment]
already merged most of it, current status:
1. still need to commit some fixes from master branch
2. one test needs to be corrected

I'll push the branch when I'm done with it and it is in a stable/working state and you can comment on it from that point. I do however want to get this going pretty quickly now that we have the momentum going!

**Comment by lirantal** [comment]
Updated 0.4.0 branch ready to be merged at: https://github.com/meanjs/mean/pull/616

**Comment by ilanbiala** [comment]
This issue has fragmented. For docs info go to #443
